source/extensions/filters/network/http_connection_manager/config.h

```diff
@@ -224,7 +224,7 @@ class HttpConnectionManagerConfig : Logger::Loggable<Logger::Id::config>,
       HttpConnectionManagerProto::OVERWRITE};
   std::string server_name_;
   Tracing::HttpTracerSharedPtr http_tracer_{std::make_shared<Tracing::HttpNullTracer>()};
-  Http::TracingConnectionManagerConfigPtr tracing_config_;
+  Http::TracingConnectionManagerConfigPtr tracing_config_ = nullptr;
```

Comment:
Anywhere where `tracingConfig()` (here: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1083 or here: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1533, and other similar places) an additional check for a nullptr needs to be made. Alternatively, `tracing_config_` here can be initialized with a null/dummy object to avoid the additional checks scattered throughout the code.

Reply:
@dmitri-d Some of those calls are already protected by an `if`. E.g., the first one you mentioned is protected [here](https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1043). Some others are internal or direct usage of that api, whereas it should be through the proper method. E.g., in your second example should go through `tracingConfig()` method, which I changed to return `null` if it's unset. Anyway, I've put some asserts in the code to prevent such bad usage. Let me know what you think.